external-crates/move/crates/move-package-alt/src/package/lockfile.rs

```diff
@@ -46,24 +45,71 @@ pub struct Publication<F: MoveFlavor + fmt::Debug> {
 }
 
 #[derive(fmt::Debug, Serialize, Deserialize)]
-#[derive_where(Default, Clone)]
-#[serde(rename_all = "kebab-case")]
+#[derive_where(Clone)]
 #[serde(bound = "")]
-struct UnpublishedTable<F: MoveFlavor + fmt::Debug> {
-    dependencies: UnpublishedDependencies<F>,
+struct DependencyInfo<F: MoveFlavor + fmt::Debug> {
+    #[serde(flatten)]
+    data: BTreeMap<PackageName, DepInfo<F>>,
+}
 
-    #[serde(default)]
-    dep_replacements: BTreeMap<EnvironmentName, UnpublishedDependencies<F>>,
+#[derive(fmt::Debug, Serialize, Deserialize)]
+#[derive_where(Clone)]
+#[serde(bound = "")]
+struct DepInfo<F: MoveFlavor + fmt::Debug> {
+    source: PinnedInfo<F>,
+    manifest_digest: String,
+    deps: BTreeMap<PackageName, PackageName>,
 }
 
+/// Defines the pinned information for a dependency. It is used to distinguish between a local
+/// dependency of a dependency, or a dependency, or define this package as the root of the
+/// dependency graph.
 #[derive(fmt::Debug, Serialize, Deserialize)]
-#[derive_where(Default, Clone)]
+#[derive_where(Clone)]
 #[serde(bound = "")]
-struct UnpublishedDependencies<F: MoveFlavor + fmt::Debug> {
-    #[serde(default)]
-    pinned: BTreeMap<PackageName, PinnedDependencyInfo<F>>,
-    #[serde(default)]
-    unpinned: BTreeMap<PackageName, ManifestDependencyInfo<F>>,
+#[serde(untagged)]
+pub enum PinnedInfo<F: MoveFlavor + fmt::Debug> {
+    /// A local dependency relative to this dependency.
+    Local(LocalPinnedDependency<F>),
+    /// A dependency that is pinned to a specific version.
+    Remote(PinnedDependencyInfo<F>),
+    /// Marks this package as the root of the dependency tree.
+    Root(Root),
+}
+
+/// Defines a package as the root of a dependency tree.
+#[derive(Clone, fmt::Debug, Serialize, Deserialize)]
+pub struct Root {
+    root: bool,
+}
+
+#[derive(fmt::Debug, Serialize, Deserialize)]
+#[derive_where(Clone)]
+#[serde(bound = "")]
+pub struct LocalPinnedDependency<F: MoveFlavor + fmt::Debug> {
+    /// The relative path to the local dependency
+    local: PathBuf,
+    /// This local's dependency parent information
+    parent: PinnedDependencyInfo<F>,
+}
+
+impl<F: MoveFlavor + fmt::Debug> DepInfo<F> {
```

Comment:
I wonder if we want to just expose the fields for this type since it's just plain old data